src/node_api.cc

```diff
@@ -1090,14 +1094,23 @@ napi_status napi_define_properties(napi_env env,
   for (size_t i = 0; i < property_count; i++) {
     const napi_property_descriptor* p = &properties[i];
 
-    v8::Local<v8::Name> name;
-    CHECK_NEW_FROM_UTF8(env, name, p->utf8name);
+    v8::Local<v8::Name> property_name;
+
+    if (p->utf8name != nullptr) {
```

Comment:
Sounds fine to me. That's probably a better place for the macros anyway. Though you'll probably also need to move up some helper functions like `napi_set_last_error()` (or just declare them).